### DISCUSSION TITLE
Instance Selection on River

### DISCUSSION THREAD

**Discussion by gintriago12** [online-ml/river#684] [Q&A] [answered]
Hi,

I want to merge my python code of the uniFied Instance Selection metHod (FISH) into river, but I am not sure which base class to useWhich base class do you recommend for this purpose?

In general, FISH avoids using all the historical dataset by providing the closest (in time and space) instances to the learning model for training.

Best,

**Comment by MaxHalford** [comment, marked as answer]
Hello @gintriago12! 

> I want to merge my python code of the uniFied Instance Selection metHod (FISH) into river

We very much appreciate you putting in the effort!

> Which base class do you recommend for this purpose?

Well it depends: what does your model do? Is it a regressor? A classifier? Or is it here to transform data in some way?

> In general, FISH avoids using all the historical dataset by providing the closest (in time and space) instances to the learning model for training.

That sounds interesting. But it doesn't per say as online learning. It sounds like some fancy way to do batch learning. Our philosophy with River is to only implement and maintain pure online learning models. Do you have a reference paper where FISH is discussed?

**Reply by gintriago12** [reply]
Hi Max,

Thanks for answering timely.  This is the link to the article that describes [FISH](http://eprints.bournemouth.ac.uk/18567/1/FISH_journal_preprint.pdf), the author is Indre Zliobaite who has some works with Prof. Bifet.

I agree, FISH has a gist of fancy batch learning. FISH, as taken from the abstract, is a method that determines an optimal training set size online at every time step using cross validation. It is a wrapper approach, it can be used by plugging in different base classifiers.

However, I think FISH shares a same characteristic with DDM (drift detection method), in the way that DDM selects a window of instances from the warning level example kw to the drift level kd. Then, such window is used to induce a new decision model. On the other hand, FISH uses a similarity function that combines space and time to select the most appropriate instances to induce a new model.

Best,
